### PR TITLE
[Testing:Developer] Don't Abandon Approved PRs

### LINKED ISSUE #10211: "Abandoned" label shouldn't be slapped on PRs that have been approved
Currently, an "Abandoned PR - Needs New Owner" label is slapped on a PR if there's no commits or comments made within 2 weeks. This shouldn't be the case for the PRs that are approved and ready to merge.

### PR DESCRIPTION
**Pull request by michaelkrar**
### Please check if the PR fulfills these requirements:

* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant
* [ ] Screenshots are attached to Github PR if visual/UI changes were made

### What is the current behavior?

Closes #10211

### What is the new behavior?
The python script will now check for whether a PR has approving reviews and has not had a changes requested review after the last approving review. It will also remove the abandoned tag if the PR is considered to be approved.
Todo: Check if it passes yml checks?
### Other information?

No, as this can only possibly reduce the amount of PRs that are marked as abandoned as it increases the criteria.

Todo: need to figure out how to test this.
There's probably a better way but this is the only thing that comes to mind: https://github.com/nektos/act?tab=readme-ov-file

### CODE REVIEW

**Comment by IDzyre** [comment]
You can create a fork, and test it on that since it is your own repository, and can create/label/approve issues as you please.

**Comment by michaelkrar** [comment]
https://github.com/michaelkrar/Submitty/pull/5
Ok so so far I have seen that the changes don't stop a PR that should be abandoned from being marked as abandoned.
Still need to test for when there is either an approving review.

**Comment by ziesski** [comment]
Please follow the title guideline, and check cypress test you are failing.

File: .github/bin/label_abandoned_prs.py [hunk 1]
```diff
@@ -36,7 +36,13 @@
...
             already_abandoned = True
 
-    if tdiff > twelve_days and not already_abandoned and not already_warned:
+    for review in json_data['reviews']:
+        if review["state"] == "APPROVED":
+            approved = True
+        if review["state"] == "CHANGES_REQUESTED":
+            approved = False
+
+    if tdiff > twelve_days and not already_abandoned and not already_warned and not approved:
```
Comment (04vmatsibekker) [inline .github/bin/label_abandoned_prs.py:45]: what if someone approves after the abandon tag is put on

File: .github/bin/label_abandoned_prs.py [hunk 2]
```diff
@@ -36,7 +36,13 @@
         if labels['name'] == 'Abandoned PR - Needs New Owner':
             already_abandoned = True
 
-    if tdiff > twelve_days and not already_abandoned and not already_warned:
+    for review in json_data['reviews']:
```
Comment (04vmatsibekker) [inline .github/bin/label_abandoned_prs.py:39]: What if multiple people review the PR and one person has changes requested but the last person clicks "Approved", making the variable approved = True. Should we then have the state be considered approved if the developer does not address the first person's requested changes?

**Review by 04vmatsibekker — DISMISSED** [review]

**Review by williamjallen — APPROVED** [review]

**Pull request merged**